src/main/com/yetanalytics/lrs/pedestal/routes.cljc

```diff
@@ -41,30 +43,16 @@
         method
         method-not-allowed
         :route-name (keyword route-name-ns (name method))]
-       (let [doc-params-interceptor
-             (xapi-i/params-interceptor
-              (case resource-tuple
-                ["activities" "state"]
-                (case method
-                  :put    :xapi.activities.state.PUT.request/params
-                  :post   :xapi.activities.state.POST.request/params
-                  :get    :xapi.activities.state.GET.request/params
-                  :head   :xapi.activities.state.GET.request/params
-                  :delete :xapi.activities.state.DELETE.request/params)
-                ["activities" "profile"]
-                (case method
-                  :put    :xapi.activities.profile.PUT.request/params
-                  :post   :xapi.activities.profile.POST.request/params
-                  :get    :xapi.activities.profile.GET.request/params
-                  :head   :xapi.activities.profile.GET.request/params
-                  :delete :xapi.activities.profile.DELETE.request/params)
-                ["agents"     "profile"]
-                (case method
-                  :put    :xapi.agents.profile.PUT.request/params
-                  :post   :xapi.agents.profile.POST.request/params
-                  :get    :xapi.agents.profile.GET.request/params
-                  :head   :xapi.agents.profile.GET.request/params
-                  :delete :xapi.agents.profile.DELETE.request/params)))
+       (let [method-kw  (case method
+                          :head :get
+                          method)
+             method-str (-> method-kw name clojure.string/upper-case)
+             token-vec  ["xapi" resource doc-type method-str "request/params"]
+             spec-kw (-> (clojure.string/join \. token-vec)
+                         keyword)
+             doc-params-interceptor 
+             (xapi-i/params-interceptor spec-kw)
+             
```

Comment:
Is this change relevant to adding instrumentation? If it's just that you need the `spec-kw` I'd prefer that you bind it separately using the old method. It was designed the way it was to be explicit/readable with the specs involved (as this app has a LOT of specs)

Reply:
reverted